openforcefield/topology/topology.py

```diff
@@ -0,0 +1,2190 @@
+        TODO: Do we want to pin ourselves to a single standard aromaticity model?
+    type : str
+        String based bond type
+    order : int
+        Integral bond order
+    fractional_bondorder : float, optional
+        Fractional bond order, or None.
+
+    """
+    def __init__(self, atom1, atom2, bondtype, fractional_bondorder=None):
+        """
+        Create a new chemical bond.
+        """
+        # TODO: Make sure atom1 and atom2 are both Atom types
+        self._atom1 = atom1
+        self._atom2 = atom2
+        self._type = bondtype
+        self._fractional_bondorder = fractional_bondorder
+
+    # TODO: add getters for each of these bond properties
+
+    @property
+    def atom1(self):
+        return self._atom1
+
+    @property
+    def atom2(self):
+        return self._atom2
+
+    @property
+    def atoms(self):
+        return (self._atom1, self._atom2)
+
+    def type(self):
+        return self._type
+
+    @property
+    def fractional_bondorder(self):
+        return self._fractional_bondorder
+
+    @fractional_bondorder.setter
+    def fractional_bondorder(self, value):
+        self._fractional_bondorder = value
+
+# =============================================================================================
+# TopologyAtom
+# =============================================================================================
+
+class TopologyAtom(Serializable):
+    """
+
+    """
+
+    def __init__(self, atom, topology_molecule):
+        """
+
+        Parameters
+        ----------
+        atom : An openforcefield.topology.molecule.Atom
+            The reference atom
+        topology_molecule : An openforcefield.topology.topology.TopologyMolecule
+            The TopologyMolecule that this TopologyAtom belongs to
+        """
+        # TODO: Type checks
+        self._atom = atom
+        self._topology_molecule = topology_molecule
+
+
+    @property
+    def atom(self):
+        """
+        Get the reference Atom for this TopologyAtom.
+
+        Returns
+        -------
+        an openforcefield.topology.molecule.Atom
+        """
+        return self._atom
+
+    @property
+    def atomic_number(self):
+        """
+        Get the atomic number of this atom
+
+        Returns
+        -------
+        int
+        """
+        return self._atom.atomic_number
+    @property
+    def topology_molecule(self):
+        """
+        Get the TopologyMolecule that this TopologyAtom belongs to.
+
+        Returns
+        -------
+        openforcefield.topology.topology.TopologyMolecule
+        """
+        return self._topology_molecule
+
+    @property
+    def molecule(self):
+        """
+        Get the reference Molecule that this TopologyAtom belongs to.
+
+        Returns
+        -------
+        openforcefield.topology.molecule.Molecule
+        """
+        return self._topology_molecule.molecule
+
+
+    @property
+    def topology_atom_index(self):
```

Comment:
Do we need to prepend these API calls with `topology_` if we know that it is a `TopologyAtom`?
We might want to tag docstrings for parts of the API under debate with warnings that specify the API is experimental and subject to change.